File: java/easy/Running_sum_of_1d_array.java
package easy;

public class Running_sum_of_1d_array {
	class Solution {
	    public int[] runningSum(int[] nums) {

	        for(int i = 1; i < nums.length; i++) {
	            nums[i] = nums[i - 1] + nums[i];
	        }

	        return nums;
	    }
	}
}

File: javascript/easy/Running_Sum_of_1d_Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
const runningSum = function (nums) {
  return nums.map((v, i) =>
    nums.slice(0, i + 1).reduce((prev, current) => prev + current, 0)
  );
};

console.log(runningSum([1, 2, 3, 4]));
